### PR TITLE
CI: add PT 1.10

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
enable PyTorch 1.10

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/workflows/ci_test-conda.yml [hunk 1]
```diff
@@ -13,8 +13,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7]
-        pytorch-version: [1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9]
+        python-version: ["3.7"]
+        pytorch-version: ["1.3", "1.4", "1.5", "1.6", "1.7", "1.8", "1.9", "1.10"]
```
Comment (carmocca) [inline .github/workflows/ci_test-conda.yml:17]: should remove 1.3?

Comment (Borda) [reply .github/workflows/ci_test-conda.yml:17]: Let's first get 1.10 on board... need to find compatible TV version

Comment (carmocca) [reply .github/workflows/ci_test-conda.yml:17]: I asked because 1.4 is our minimum supported version

Comment (Borda) [reply .github/workflows/ci_test-conda.yml:17]: yes, I know that PL min is 1.4, but till we crash to some limitations we can preserve it...

**Comment by Borda** [comment]
requested https://github.com/pytorch/vision/issues/4117

File: .github/workflows/ci_test-conda.yml [hunk 2]
```diff
@@ -62,11 +62,11 @@ jobs:
         conda list
         pip --version
         python ./requirements/adjust-versions.py requirements.txt
-        python ./requirements/adjust-versions.py requirements/image.txt
+        python ./.github/prune-packages.py requirements/image.txt torchvision
         pip install --requirement requirements.txt --upgrade-strategy only-if-needed --quiet
-        pip install --requirement requirements/test.txt --upgrade-strategy only-if-needed --quiet --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
+        pip install --requirement requirements/test.txt --upgrade-strategy only-if-needed --quiet
```
Comment (carmocca) [inline .github/workflows/ci_test-conda.yml:67]: Just so you know `--upgrade-strategy only-if-needed` is the default, no need to set it 🌵

File: requirements/adjust-versions.py [hunk 3]
```diff
@@ -5,7 +5,8 @@
 from typing import Dict, Optional
 
 VERSIONS = [
-    dict(torch="1.9.0", torchvision="0.10.0", torchtext=""),  # nightly
+    dict(torch="1.10.0", torchvision="0.11.0", torchtext=""),  # nightly
+    dict(torch="1.9.0", torchvision="0.10.0", torchtext=""),
```
Comment (carmocca) [inline requirements/adjust-versions.py:9]: why don't you set the torchtext version?

Comment (Borda) [reply requirements/adjust-versions.py:9]: it is not available I guess https://github.com/pytorch/text#installation

File: .github/workflows/ci_test-conda.yml [hunk 4]
```diff
@@ -62,11 +62,11 @@ jobs:
         conda list
         pip --version
         python ./requirements/adjust-versions.py requirements.txt
-        python ./requirements/adjust-versions.py requirements/image.txt
+        python ./.github/prune-packages.py requirements/image.txt torchvision
         pip install --requirement requirements.txt --upgrade-strategy only-if-needed --quiet
```
Comment (Borda) [inline .github/workflows/ci_test-conda.yml:66]:
```suggestion
        pip install --requirement requirements.txt --quiet
```

**Review by awaelchli — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**